lib/blocs/preferences_cubit.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:async';
+
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:trufi_core/models/preferences.dart';
+import 'package:trufi_core/repository/local_repository.dart';
+import 'package:trufi_core/trufi_models.dart';
+import 'package:uuid/uuid.dart';
+
+class PreferencesCubit extends Cubit<Preference> {
+  LocalRepository localRepository;
+  static const bool defaultOnline = true;
+  static const String defaultMapType = MapStyle.streets;
+  static const String defaultLanguageCode = "en";
+
+  PreferencesCubit(this.localRepository)
+      : super(const Preference(defaultLanguageCode, "", defaultMapType,
+            loadOnline: defaultOnline)) {
+    load();
+  }
+
+  void updateMapType(String mapType) {
+    emit(state.copyWith(currentMapType: mapType));
+  }
+
+  Future<void> updateLanguage(String languageCode) async {
+    localRepository.saveLanguageCode(languageCode);
+    emit(state.copyWith(languageCode: languageCode));
+  }
+
+  Future<void> updateOnline({bool loadOnline = false}) async {
+    localRepository.saveUseOnline(loadOnline: loadOnline);
+    emit(state.copyWith(loadOnline: loadOnline));
+  }
+
+  Future<void> updateStateHomePage(String stateHomePage) async {
+    if (stateHomePage == null) {
+      await localRepository.deleteStateHomePage();
+    } else {
+      await localRepository.saveStateHomePage(stateHomePage);
+    }
+    emit(state.copyWith(stateHomePage: stateHomePage));
```

Comment:
when stateHomePage equals null, copyWith receives a null parameter and then the "emit" is called with the same state. Is this correct behavior ?, or do we need to emit a state like "DeleteStateHomePage"?

Reply:
Regarding if it is the `same` state, I guess not. `stateHomePage` comes as a parameter into the function, and if it would be null, the new state will be `null` for the stateHomePage`.

Good question. Regarding the code, I tried to keep it, for the most part, as close to the original as possible. It got deleted in a lot of places by setting a null value. Actually, I guess at the moment I removed the null possibility from the HomePageState. We have to keep an eye on that if that has any bad side effects.

Reply:
Great, I can do "manual tstiong" today 👍 .

Reply:
I did "manual tests" for the main trufi features, I found no critical bugs, only this.
https://github.com/trufi-association/trufi-core/pull/420#issuecomment-824842127

https://user-images.githubusercontent.com/42806689/115804675-79a00080-a3b1-11eb-982f-69b49b79ca8e.mov

Reply:
I did "manual testing" in iOS too, I found no critical bugs, only this.
#420 (comment).

Reply:
I found a bug(IOS and Android) in the "Your Places" page, when I am clicking a place it does not search my route.

https://user-images.githubusercontent.com/42806689/115813154-a576b280-a3c0-11eb-80f8-3983d7b24997.mov